youtube_dl/extractor/myspace.py

```diff
@@ -204,3 +204,46 @@ def _real_extract(self, url):
             'title': self._og_search_title(webpage),
             'entries': entries,
         }
+
+
+class MySpaceArtistSongsIE(InfoExtractor):
+    IE_NAME = 'MySpace:artist:songs'
+    _VALID_URL = r'https?://myspace\.com/(?P<artist>[^/]*)/music/songs'
+
+    _TESTS = [{
+        'url': 'https://myspace.com/starset2/music/album/transmissions-19455773',
+        'url': 'https://myspace.com/ryanlewismusic/music/songs',
+        'info_dict': {
+            'title': 'RYAN LEWIS',
+        },
+        'playlist_count': 8,
+    }, {
+        'url': 'https://myspace.com/studio99/music/songs/',
+        'info_dict': {
+            'title': 'Studio 99 IS CLOSED!! R.I.P.',
+        },
+        'playlist_count': 4,
+    }]
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        display_id = mobj.group('artist')
+        webpage = self._download_webpage(url, display_id)
+        tracks_paths = re.findall(r'<meta itemprop="url" content ="(.*?)">', webpage)
+        if not tracks_paths:
+            raise ExtractorError(
+                '%s: No songs found, try using proxy' % display_id,
```

Comment:
Well. This IE is copy-based from the Album one (MySpaceAlbumIE). I just copied and adapted the code artist instead album. I guess the original author put this due geoblocking giving different results. If change this part, it's better to change the other too, to keep coherent.

Commits to this part:
95c673a148f1a6cd8571b4dfa931d98092500fc2